homeassistant/components/zwave/api.py

```diff
@@ -0,0 +1,112 @@
+"""API class to give info to the Z-Wave panel."""
+
+import logging
+import homeassistant.core as ha
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import HTTP_NOT_FOUND
+from . import const
+
+_LOGGER = logging.getLogger(__name__)
+
+ZWAVE_NETWORK = 'zwave_network'
+
+
+class ZWaveNodeGroupView(HomeAssistantView):
+    """View to return the nodes group configuration."""
+
+    url = "/api/zwave/groups/{node_id}"
+    name = "api:zwave:groups"
+
+    @ha.callback
+    def get(self, request, node_id):
+        """Retrieve groups of node."""
+        # pylint: disable=import-error
+        from openzwave.group import ZWaveGroup
+
+        hass = request.app['hass']
+        network = hass.data.get(ZWAVE_NETWORK)
+        _LOGGER.info(network.nodes.get(int(node_id)))
+        node = network.nodes.get(int(node_id))
+        if node is None:
+            return self.json_message('Node not found', HTTP_NOT_FOUND)
+        groupdata = node.groups_to_dict()
+        groups = {}
+        for key in groupdata.keys():
+            groupnode = ZWaveGroup(key, network, int(node_id))
+            groups[key] = {'associations': groupnode.associations,
+                           'association_instances':
+                           groupnode.associations_instances,
+                           'label': groupnode.label,
+                           'max_associations': groupnode.max_associations}
+        _LOGGER.info('Groups: %s', groups)
+        if groups:
+            return self.json(groups)
+        else:
+            return self.json_message('Node not found', HTTP_NOT_FOUND)
```

Comment:
If there are no groups, you should not return a NOT FOUND. Instead return the empty object. That way the client knows that the node exists but it has no groups.